src/lib/comms/sol-lwm2m.c

```diff
@@ -3085,6 +3163,127 @@ handle_create(struct sol_lwm2m_client *client,
+                    SOL_WRN("Bootstrap Write on Object Instance /%"
+                        PRIu16 "/%" PRIu16 " failed!", obj_ctx->obj->id, instance_tlvs->id);
+                }
+            }
+
+            SOL_DBG("Bootstrap Write on Object /%" PRIu16 " succeeded!", obj_ctx->obj->id);
+
+            return SOL_COAP_RESPONSE_CODE_CHANGED;
+        } else {
+            SOL_WRN("Only TLV is supported for writing multiple Object Instances."
+                " Received: %" PRIu16 "", payload.type);
+            return SOL_COAP_RESPONSE_CODE_UNSUPPORTED_CONTENT_FORMAT;
+        }
+
+    } else if (path_size < 3 && client->is_bootstrapping) {
+        //Bootstrap Write on Object Instance (e.g.: PUT /1/5)
+        //In this case the payload is composed of <multiple TLVs> of type MULTIPLE_RESOURCES
+        // or RESOURCE_WITH_VALUE and each TLV of type MULTIPLE_RESOURCES can contain
+        // multiple TLVs of type RESOURCE_INSTANCE
+        if (payload.type == SOL_LWM2M_CONTENT_TYPE_TLV) {
+            //Instance already exists?
+            if (obj_instance) {
+                r = write_instance_tlv_or_resource(client, obj_ctx, obj_instance, -1, payload);
+            } else {
+                r = handle_create(client, obj_ctx, path[1], payload);
+            }
+
+            if (r == SOL_COAP_RESPONSE_CODE_CHANGED || r == SOL_COAP_RESPONSE_CODE_CREATED) {
+                SOL_DBG("Bootstrap Write on Object Instance /%"
+                    PRIu16 "/%" PRIu16 " succeeded!", obj_ctx->obj->id, path[1]);
+                return SOL_COAP_RESPONSE_CODE_CHANGED;
+            } else {
+                return r;
+            }
+        } else {
+            SOL_WRN("Only TLV is supported for writing Object Instance."
+                " Received: %" PRIu16 "", payload.type);
+            return SOL_COAP_RESPONSE_CODE_UNSUPPORTED_CONTENT_FORMAT;
+        }
+
+    } else if (client->is_bootstrapping) {
+        //Bootstrap Write on Resource (e.g.: PUT /1/5/1)
+        //In this case the payload is composed of a <single TLV> of type MULTIPLE_RESOURCES
+        // or RESOURCE_WITH_VALUE and in case of TLV of type MULTIPLE_RESOURCES this TLV can contain
+        // multiple TLVs of type RESOURCE_INSTANCE
+        r = write_instance_tlv_or_resource(client, obj_ctx, obj_instance, path[2], payload);
+
+        if (r == SOL_COAP_RESPONSE_CODE_CHANGED || r == SOL_COAP_RESPONSE_CODE_CREATED) {
+            SOL_DBG("Bootstrap Write on Resource /%" PRIu16 "/%" PRIu16 "/%" PRIu16 " succeeded!",
+                obj_ctx->obj->id, obj_instance->id, path[2]);
+            return SOL_COAP_RESPONSE_CODE_CHANGED;
+        } else {
+            return r;
+        }
+
+    } else {
+        //Management Write as [Replace] on Object Instance (e.g.: PUT /1/5) or Resource (e.g.: PUT /1/5/1)
+        // or as [Partial Update] on Object Instance (e.g.: POST /1/5)
+        return write_instance_tlv_or_resource(client, obj_ctx, obj_instance, path[2], payload);
```

Comment:
Another awesome catch haha.

Changed to:
```C
    } else {
        //{Bootstrap Write on Resource (e.g.: PUT /1/5/1)
        //In this case the payload is composed of a <single TLV> of type MULTIPLE_RESOURCES
        // or RESOURCE_WITH_VALUE and in case of TLV of type MULTIPLE_RESOURCES this TLV can contain
        // multiple TLVs of type RESOURCE_INSTANCE} or
        //{Management Write as [Replace] on Object Instance (e.g.: PUT /1/5) or Resource (e.g.: PUT /1/5/1)
        // or as [Partial Update] on Object Instance (e.g.: POST /1/5)}
        r = write_instance_tlv_or_resource(client, obj_ctx, obj_instance, path[2], payload);

        if (r == SOL_COAP_RESPONSE_CODE_CHANGED || r == SOL_COAP_RESPONSE_CODE_CREATED) {
            SOL_DBG("Bootstrap/Management Write on Resource /%"
                PRIu16 "/%" PRIu16 "/%" PRIu16 " succeeded!",
                obj_ctx->obj->id, obj_instance->id, path[2]);
            return SOL_COAP_RESPONSE_CODE_CHANGED;
        } else {
            SOL_WRN("Bootstrap/Management Write on Resource /%"
                PRIu16 "/%" PRIu16 "/%" PRIu16 " failed!",
                obj_ctx->obj->id, obj_instance->id, path[2]);
            return r;
        }
    }
```